bluemira/equilibria/optimisation/harmonics/harmonics_approx_functions.py

```diff
@@ -285,6 +294,37 @@ def collocation_points(
         collocation_r = np.sqrt(collocation_x**2 + collocation_z**2)
         collocation_theta = np.arctan2(collocation_x, collocation_z)
 
+    if point_type is PointType.GRID_POINTS:
+        # Create uniform, rectangular grid using max and min LCFS values
+        grid_num_x, grid_num_z = grid_num
+        rect_grid = Grid(
+            np.amin(x_bdry),
+            np.amax(x_bdry),
+            np.amin(z_bdry),
+            np.amax(z_bdry),
+            nx=grid_num_x,
+            nz=grid_num_z,
+        )
+
+        # Only use grid points that are within LCFS
+        collocation_x = np.array([])
+        collocation_z = np.array([])
+        for i in np.arange(grid_num_x):
+            for j in np.arange(grid_num_x):
+                in_poly = in_polygon(
+                    x=rect_grid.x[i, j],
+                    z=rect_grid.z[i, j],
+                    poly=plasma_boundary.xz.T,
+                    include_edges=True,
```

Comment:
As you are setting up your arrays yourself here, you could consider using `geometry::coordinates.py::polygon_in_polygon` or perhaps more canonically `equilibria::find.py::in_plasma/in_zone` if speed is an issue; you'd get the JIT on the for loops in that case. It's probably not necessary, as this looks just fine and is nice and legible. Just if it was going to be in some constraint or objective function it might be better.